src/QsCompiler/TextProcessor/SyntaxBuilder.fs

```diff
@@ -329,7 +329,11 @@ let internal buildTupleItem validSingle bundle errCode missingCode fallback cont
 /// In order to guarantee correct whitespace management, the name needs to be parsed as a term.
 let internal symbolNameLike errCode = 
     let identifier = 
-        let id = IdentifierOptions(isAsciiIdStart = isSymbolStart, isAsciiIdContinue = isSymbolContinuation) |> identifier
+        let id = IdentifierOptions(
```

Comment:
Ah, so this actually worked; by adding it to the precheck?

Reply:
Yes. Unsupported symbols aren't treated as identifiers. If you have unsupported symbol where identifier is supposed to be you'll get "expected identifier".

---
In reply to: [442525499](https://github.com/microsoft/qsharp-compiler/pull/467#discussion_r442525499) [](ancestors = 442525499)